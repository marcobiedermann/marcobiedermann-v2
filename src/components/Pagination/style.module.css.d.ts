/* eslint-disable @typescript-eslint/camelcase */

export const pagination: string;
export const pagination__link: string;
