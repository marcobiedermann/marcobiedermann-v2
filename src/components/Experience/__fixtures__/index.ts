const experience = {
  company: 'Company name',
  endDate: new Date('2001-01-01'),
  location: 'Berlin, DE',
  id: '71448e3e-65d9-45cb-8e95-50e1761c2f56',
  startDate: new Date('2000-01-01'),
  title: 'Job title',
  url: 'http://www.example.com',
};

export default experience;
