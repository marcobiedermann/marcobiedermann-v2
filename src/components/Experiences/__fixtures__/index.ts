import experience from '../../Experience/__fixtures__';

const experiences = [experience];

export default experiences;
