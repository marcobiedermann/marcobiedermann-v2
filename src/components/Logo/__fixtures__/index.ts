const project = {
  id: '1',
  logo: {
    file: {
      url: 'https://via.placeholder.com/150x50',
    },
    title: 'Logo Title',
  },
};

export default project;
