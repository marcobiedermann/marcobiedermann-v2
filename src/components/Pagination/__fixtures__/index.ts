const next = {
  id: 'next',
  name: 'Next',
  path: '/next',
};

const previous = {
  id: 'previous',
  name: 'Previous',
  path: '/previous',
};

export { next, previous };
