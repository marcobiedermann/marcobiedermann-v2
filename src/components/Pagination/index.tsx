import classNames from 'classnames';
import React from 'react';
import Route, { RouteProps } from '../Route';
import styles from './style.module.css';

export interface PaginationProps {
  className?: string;
  next?: RouteProps;
  previous?: RouteProps;
  routes: RouteProps[];
}

const Pagination: React.FC<PaginationProps> = (props) => {
  const { className, next, previous, routes, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.pagination)} {...otherProps}>
      {previous && (
        <li>
          <Route className={styles.pagination__link} {...previous} rel="previous" />
        </li>
      )}
      {routes.map((route) => (
        <li key={route.id}>
          <Route className={styles.pagination__link} {...route} />
        </li>
      ))}
      {next && (
        <li>
          <Route className={styles.pagination__link} {...next} rel="next" />
        </li>
      )}
    </ul>
  );
};

export default Pagination;
