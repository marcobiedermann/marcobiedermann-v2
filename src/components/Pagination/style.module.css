.pagination {
  display: grid;
  gap: 0.5em;
  grid-auto-flow: column;
  justify-content: center;
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 1.5em;
  padding: 0;
}

.pagination__link {
  border: 1px solid;
  display: inline-block;
  min-width: 2.33em;
  padding: 0.25em 0.5em;
  text-align: center;
}
