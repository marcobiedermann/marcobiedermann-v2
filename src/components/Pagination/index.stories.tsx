/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Pagination from '.';
import { next, previous } from './__fixtures__';

storiesOf('Pagination', module).add('default', () => <Pagination next={next} previous={previous} routes={[]} />);
