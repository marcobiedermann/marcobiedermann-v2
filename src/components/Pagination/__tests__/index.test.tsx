/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Pagination from '..';
import route from '../../Route/__fixtures__';
import { next, previous } from '../__fixtures__';

describe('Pagination component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Pagination routes={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render routes correctly', () => {
    const routes = [route];
    const { asFragment } = render(<Pagination next={next} previous={previous} routes={routes} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
