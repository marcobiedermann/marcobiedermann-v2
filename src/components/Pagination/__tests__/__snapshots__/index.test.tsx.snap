// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  />
</DocumentFragment>
`;

exports[`Pagination component should render routes correctly 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li>
      <a
        class="pagination__link"
        href="/previous"
        id="previous"
        rel="previous"
      >
        Previous
      </a>
    </li>
    <li>
      <a
        class="pagination__link"
        href="/path/to/page"
        id="b6c3ceb6-d89f-4a03-9e96-77768f278909"
      >
        Page name
      </a>
    </li>
    <li>
      <a
        class="pagination__link"
        href="/next"
        id="next"
        rel="next"
      >
        Next
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
